epoxy-adapter/src/main/java/com/airbnb/epoxy/WrappedEpoxyModelCheckedChangeListener.java

```diff
@@ -0,0 +1,102 @@
+package com.airbnb.epoxy;
+
+import android.view.View;
+import android.view.ViewParent;
+import android.widget.CompoundButton;
+import android.widget.CompoundButton.OnCheckedChangeListener;
+
+import androidx.annotation.Nullable;
+import androidx.recyclerview.widget.RecyclerView;
+import androidx.recyclerview.widget.RecyclerView.ViewHolder;
+
+/**
+ * Used in the generated models to transform normal checked change listener to model
+ * checked change.
+ */
+public class WrappedEpoxyModelCheckedChangeListener<T extends EpoxyModel<?>, V>
+    implements OnCheckedChangeListener {
+
+  private final OnModelCheckedChangeListener<T, V> originalCheckedChangeListener;
+
+  public WrappedEpoxyModelCheckedChangeListener(
+      OnModelCheckedChangeListener<T, V> checkedListener
+  ) {
+    if (checkedListener == null) {
+      throw new IllegalArgumentException("Checked change listener cannot be null");
+    }
+
+    this.originalCheckedChangeListener = checkedListener;
+  }
+
+  @Override
+  public void onCheckedChanged(CompoundButton button, boolean isChecked) {
+    EpoxyViewHolder epoxyHolder = getEpoxyHolderForChildView(button);
+    if (epoxyHolder == null) {
+      throw new IllegalStateException("Could not find RecyclerView holder for clicked view");
+    }
+
+    final int adapterPosition = epoxyHolder.getAdapterPosition();
+    if (adapterPosition != RecyclerView.NO_POSITION) {
+      originalCheckedChangeListener
+          .onChecked((T) epoxyHolder.getModel(), (V) epoxyHolder.objectToBind(), button,
+          isChecked, adapterPosition);
+    }
+  }
+
+  @Nullable
+  private static EpoxyViewHolder getEpoxyHolderForChildView(View v) {
```

Comment:
these functions could be pulled out to util functions and shared with the click listener class

Reply:
Yes, do you want me to do it? or is it just a reminder for the future?

Reply:
If you could do it that would be great, otherwise it can be done some other time

Reply:
Done, I've called the file ListenersUtils since there was already a Utils in the same namespace, hope it's ok... otherwise I'll change it later, just let me know

Reply:
perfect, thanks!